common/lfdemod.c

```diff
@@ -426,543 +308,410 @@ uint32_t manchesterEncode2Bytes(uint16_t datain) {
 
 //by marshmellow
 //encode binary data into binary manchester 
-int ManchesterEncode(uint8_t *BitStream, size_t size)
-{
-	size_t modIdx=20000, i=0;
+//NOTE: BitStream must have double the size available in memory to do the swap
+int ManchesterEncode(uint8_t *BitStream, size_t size) {
+	size_t modIdx=size, i=0;
 	if (size>modIdx) return -1;
```

Comment:
Good catch.  Looking closer there is another problem.  I'll fix this before we merge.